nix/distribution.nix

```diff
@@ -1,30 +1,32 @@
 {
   inputs,
   self,
-  version,
   ...
 }: {
   perSystem = {pkgs, ...}: {
-    packages = {
-      inherit (pkgs) prismlauncher-qt5-unwrapped prismlauncher-qt5 prismlauncher-unwrapped prismlauncher;
-      default = pkgs.prismlauncher;
+    packages = let
+      ourPackages = self.overlays.default pkgs null;
+    in {
+      inherit (ourPackages) prismlauncher-qt5-unwrapped prismlauncher-qt5 prismlauncher-unwrapped prismlauncher;
+      default = ourPackages.prismlauncher;
     };
   };
 
   flake = {
-    overlays.default = final: prev: let
-      # Helper function to build prism against different versions of Qt.
-      mkPrism = qt:
-        qt.callPackage ./package.nix {
-          inherit (inputs) libnbtplusplus;
-          inherit (prev.darwin.apple_sdk.frameworks) Cocoa;
-          inherit self version;
-        };
-    in {
-      prismlauncher-qt5-unwrapped = mkPrism final.libsForQt5;
-      prismlauncher-qt5 = prev.prismlauncher-qt5.override {prismlauncher-unwrapped = final.prismlauncher-qt5-unwrapped;};
-      prismlauncher-unwrapped = mkPrism final.qt6Packages;
-      prismlauncher = prev.prismlauncher.override {inherit (final) prismlauncher-unwrapped;};
+    overlays.default = final: _: let
+      version = builtins.substring 0 8 self.lastModifiedDate;
+
+      # common args for prismlauncher evaluations
+      unwrappedArgs = {
```

Comment:
imo changing this from a function to a lambda creates a bit more code duplication while not really helping readability much

Reply:
I felt like this was the cleanest way as before it inherited `prev` (or now `final`) from its context. I like to keep functions mostly self contained. The only thing that's duplicated now is the `callPackage` really